### PR TITLE
Fix Travis CI builds & drop Python 2.6 support

### PR DESCRIPTION
**Pull request by sphuber**
While addressing issue #1099 I found that the CI builds on Travis were severely broken. I took the liberty to fix them which consisted of moving to Xenial to get the required python versions and addressing various flake8 messages. I dropped 2.6 because it is difficult to install and it has been EOL for over 5 years.

### CODE REVIEW

File: .coveralls.sh [hunk 1]
```diff
@@ -1,5 +0,0 @@
-if [ "$TOX_ENV" = "py27" ]
-then
-   pip install coveralls
-   coveralls
-fi
```
Comment (k4nar) [inline .coveralls.sh]: Did we have issues with Coveralls?

**Comment by sphuber** [comment]
It was originally being run as a separate job on py2.7 but I moved it to run at the end for all jobs. Just in case there are python version specific code paths. This build for example didn't have an actual difference but the next does. Seey other PR. You can also looking at the end of one of jobs in the build to see that it actually run. I can put it back as a separate step he way it was before if you prefer.

**Comment by sphuber** [comment]
@k4nar let me know if you'd like me to revert the coveralls change. Once this is merged, I can rebase PR #1111 which will fix a quite blocking problem for a lot of people.

**Comment by k4nar** [comment]
Thank you very much for taking the time to do this!

**Comment by sphuber** [comment]
You are very much welcome, thanks for merging. With this merged PR #1111 is now ready for review as well.

**Pull request merged**